workers_launch_template.tf

```diff
@@ -0,0 +1,103 @@
+    ),
+    local.asg_tags)
+  }"]
+
+  lifecycle {
+    ignore_changes = ["desired_capacity"]
+  }
+}
+
+resource "aws_launch_template" "workers_launch_template" {
+  name_prefix = "${aws_eks_cluster.this.name}-${lookup(var.worker_groups_launch_template[count.index], "name", count.index)}"
+
+  network_interfaces {
+    associate_public_ip_address = "${lookup(var.worker_groups_launch_template[count.index], "public_ip", local.workers_group_defaults["public_ip"])}"
+    security_groups             = ["${local.worker_security_group_id}", "${var.worker_additional_security_group_ids}", "${compact(split(",",lookup(var.worker_groups_launch_template[count.index],"additional_security_group_ids", local.workers_group_defaults["additional_security_group_ids"])))}"]
+  }
+
+  iam_instance_profile = {
+    name = "${element(aws_iam_instance_profile.workers_launch_template.*.name, count.index)}"
+  }
+
+  image_id      = "${lookup(var.worker_groups_launch_template[count.index], "ami_id", local.workers_group_defaults["ami_id"])}"
+  instance_type = "${lookup(var.worker_groups_launch_template[count.index], "instance_type", local.workers_group_defaults["instance_type"])}"
+  key_name      = "${lookup(var.worker_groups_launch_template[count.index], "key_name", local.workers_group_defaults["key_name"])}"
+  user_data     = "${base64encode(element(data.template_file.launch_template_userdata.*.rendered, count.index))}"
+  ebs_optimized = "${lookup(var.worker_groups_launch_template[count.index], "ebs_optimized", lookup(local.ebs_optimized, lookup(var.worker_groups_launch_template[count.index], "instance_type", local.workers_group_defaults["instance_type"]), false))}"
+
+  monitoring {
+    enabled = "${lookup(var.worker_groups_launch_template[count.index], "enable_monitoring", local.workers_group_defaults["enable_monitoring"])}"
+  }
+
+  placement {
+    tenancy = "${lookup(var.worker_groups_launch_template[count.index], "placement_tenancy", local.workers_group_defaults["placement_tenancy"])}"
+  }
+
+  count = "${var.worker_group_launch_template_count}"
+
+  lifecycle {
+    create_before_destroy = true
+  }
+
+  block_device_mappings {
+    device_name = "${data.aws_ami.eks_worker.root_device_name}"
+
+    ebs {
+      volume_size           = "${lookup(var.worker_groups_launch_template[count.index], "root_volume_size", local.workers_group_defaults["root_volume_size"])}"
+      volume_type           = "${lookup(var.worker_groups_launch_template[count.index], "root_volume_type", local.workers_group_defaults["root_volume_type"])}"
+      iops                  = "${lookup(var.worker_groups_launch_template[count.index], "root_iops", local.workers_group_defaults["root_iops"])}"
+      delete_on_termination = true
+    }
+  }
+}
+
+resource "aws_iam_instance_profile" "workers_launch_template" {
+  name_prefix = "${aws_eks_cluster.this.name}"
```

Comment:
Will this name always be unique if someone is using both types of worker groups?

Reply:
I'm not completely sure what you mean by this.

Reply:
No worries, I think what you have now is good 🙂